cob_twist_controller/include/cob_twist_controller/constraint_solvers/solvers/constraint_solver_base.h

```diff
@@ -0,0 +1,91 @@
+/*!
+ *****************************************************************
+ * \file
+ *
+ * \note
+ *   Copyright (c) 2015 \n
+ *   Fraunhofer Institute for Manufacturing Engineering
+ *   and Automation (IPA) \n\n
+ *
+ *****************************************************************
+ *
+ * \note
+ *   Project name: care-o-bot
+ * \note
+ *   ROS stack name: cob_control
+ * \note
+ *   ROS package name: cob_twist_controller
+ *
+ * \author
+ *   Author: Marco Bezzon, email: Marco.Bezzon@ipa.fraunhofer.de
+ *
+ * \date Date of creation: March, 2015
+ *
+ * \brief
+ *   This header contains the interface description of constraint solvers
+ *   Pure virtual methods have to be implemented in subclasses
+ *
+ ****************************************************************/
+#ifndef CONSTRAINT_SOLVER_BASE_H_
+#define CONSTRAINT_SOLVER_BASE_H_
+
+#include <Eigen/Core>
+#include <Eigen/SVD>
+#include <kdl/jntarray.hpp>
+#include "cob_twist_controller/augmented_solver_data_types.h"
+
+/// Base class for solvers, defining interface methods.
+class ConstraintSolver
+{
+    public:
+        static const double DAMPING_LIMIT = 1.0e-9; ///< const. value for zero comparison with damping factor
+
+        /**
+         * TODO: Check VectorXd for q_dot?
+         * The interface method to solve the inverse kinematics problem. Has to be implemented in inherited classes.
+         * @param inCartVelocities The input velocities vector (in cartesian space).
+         * @param q The current joint positions.
+         * @param last_q_dot The last joint velocities.
+         * @return The calculated new joint velocities.
+         */
+        virtual Eigen::MatrixXd solve(const Eigen::VectorXd &inCartVelocities, const KDL::JntArray& q, const KDL::JntArray& last_q_dot) const = 0;
+
+        /**
+         * Inline method to set the damping factor
+         * @param damping The new damping factor
+         */
+        inline void setDampingFactor(double damping)
+        {
+            this->dampingFactor_ = damping;
+        }
```

Comment:
No there is no intention to provide such a function for the constraint_solver_base. 

The constraint solver base is responsible for solving the IK problem in a more general way. 

To set / add weightings it is desired to use / inherit from WeightedLeastNormSolver class. It provides a method "calculateWeighting" where a specific weighting can be implemented. Currently there is WLN_JointLimitAvoidanceSolver which implements a weighting matrix according to Chan paper ISSN 1042-296X [Page 288] (Bachelor thesis Francisco)